source/megengine.js

```diff
@@ -3,30 +3,96 @@
 
 var megengine = megengine || {};
 var base = base || require('./base');
+var zip = zip || require('./zip');
 
 megengine.ModelFactory = class {
 
     match(context) {
+        megengine.modelType = '';
+        const identifier = context.identifier;
+        const extension = identifier.split('.').pop().toLowerCase();
+        const stream = context.stream;
+        if (extension === 'mge' && stream.length >= 8) {
+            const buffer = stream.peek(Math.min(32, stream.length));
+            if (buffer.length >= 12) {
+                const magic = String.fromCharCode(...buffer.slice(8, 12));
+                if (magic === 'mgv2') {
```

Comment:
Since multiple formats used the `.mge` extension, handle those formats correctly so users trying to load other `.mge` formats will not see cryptic errors.

Reply:
done

Reply:
Duplicate of [r962172286](https://github.com/lutzroeder/netron/pull/956#discussion_r962172286).